src/ip/Address.cc

```diff
@@ -683,32 +677,73 @@ Ip::Address::InitAddr(struct addrinfo *&ai)
     }
 
     // remove any existing data.
-    if (ai->ai_addr) delete ai->ai_addr;
-
-    ai->ai_addr = (struct sockaddr*)new sockaddr_in6;
-    memset(ai->ai_addr, 0, sizeof(struct sockaddr_in6));
-
-    ai->ai_addrlen = sizeof(struct sockaddr_in6);
+    FreeAddrMember(*ai);
```

Comment:
We can also move this call into a new `else` clause for the above addrinfo allocation `if` statement.